packages/x-charts/src/ChartsLegend/ChartsLegend.tsx

```diff
@@ -4,35 +4,32 @@ import useSlotProps from '@mui/utils/useSlotProps';
 import composeClasses from '@mui/utils/composeClasses';
 import { useThemeProps, useTheme, Theme } from '@mui/material/styles';
 import { getSeriesToDisplay } from './utils';
-import { ChartsLegendClasses, getLegendUtilityClass } from './chartsLegendClasses';
+import { getLegendUtilityClass } from './chartsLegendClasses';
 import { DefaultizedProps } from '../models/helpers';
 import { DefaultChartsLegend, LegendRendererProps } from './DefaultChartsLegend';
 import { useDrawingArea } from '../hooks';
 import { useSeries } from '../hooks/useSeries';
 import { LegendPlacement } from './legend.types';
 
+export type ChartsLegendPropsBase = Omit<
+  LegendRendererProps,
+  keyof LegendPlacement | 'series' | 'seriesToDisplay' | 'drawingArea'
```

Comment:
Was this intended to make `seriesToDisplay` an invalid key to `BarChart.slotProps.legend` ?
![image](https://github.com/user-attachments/assets/8f0377e2-625c-40a7-b165-a24fddbd3600)

Reply:
It was intentional for the `ChartsLegendPropsBase` because those props are also used for [color map legends](https://mui.com/x/react-charts/legend/#continuous-color-mapping)

But not sure it was intentional for the slot to lose it @JCQuintas can you confirm?

Reply:
🤔 It wasn't intentional, but I expected our type generating scripts to pick it up 😓 

@alexfauquette